probe-rs/src/debug/variable.rs

```diff
@@ -813,95 +809,96 @@ impl Variable {
-                                                self.type_name,
+                                            // Treat this structure as a `struct`
+
+                                            if show_name {
+                                                pre_fix = Some(format!(
+                                                    "{}{:\t<indentation$}{}: {} = {} {{",
+                                                    line_feed,
+                                                    "",
+                                                    self.name,
+                                                    self.type_name,
+                                                    self.type_name,
+                                                ));
+                                            } else {
+                                                pre_fix = Some(format!(
+                                                    "{}{:\t<indentation$}{} {{",
+                                                    line_feed, "", self.type_name,
+                                                ));
+                                            }
+                                            post_fix = Some(format!(
+                                                "{}{:\t<indentation$}}}",
+                                                line_feed, ""
                                             ));
                                         }
-                                        post_fix =
-                                            Some(format!("{}{:\t<indentation$}}}", line_feed, ""));
-                                    }
-                                };
-                                if let Some(pre_fix) = &pre_fix {
-                                    compound_value = format!("{compound_value}{pre_fix}");
-                                };
+                                    };
+                                    if let Some(pre_fix) = &pre_fix {
+                                        compound_value = format!("{compound_value}{pre_fix}");
+                                    };
+                                }
+
+                                let print_name = !is_tuple;
+
+                                if child_count == children.len() {
+                                    // Do not add a separator at the end of the list
+                                    compound_value = format!(
+                                        "{}{}",
+                                        compound_value,
+                                        child.formatted_variable_value(
+                                            variable_cache,
+                                            indentation + 1,
+                                            print_name
+                                        )
+                                    );
+                                } else {
+                                    compound_value = format!(
```

Comment:
You can make the separator an interpolated variable and do an if/else that just puts the separator or an empty string instead of duplicating the entire code :)

Reply:
Done (in two places) :)